### DISCUSSION TITLE
Docker Image build and QSV support issues and arm-dependencies issue

### DISCUSSION THREAD

**Discussion by DZaremba14600** [automatic-ripping-machine/automatic-ripping-machine#1115] [General]
About 6 months ago, I posted about how I had polished a script that facilitated installing ARM onto Ubuntu 22.04 so that ARM would work without having to use it in a Docker container.  It seems that there were some who found that effort useful.

The primary reason I did that was to allow for QSV support in Handbrake for the transcoding.  I could get ARM going in a Docker container no problem, but no matter what I tried, I could never get it to recognize and use QSV.  Hence my efforts toward a non-Docker container ARM, a 'thick client install', as it were.

My goal was always to return to using ARM as a Docker Container.

Last week, I saw some follow up comments from ShitWolfyMakes related to Ver3, docker-compose support and so on, and thought I need to get back to the Docker container since that is the direction and I need to be on board.

I created a Docker container based on the latest official image from Docker Hub.  It works fine, except there is still no QSV support.  If I attach to the container I can run Handbrake-CLI and I can see that QSV is still not supported.

I figured I'd build the ARM Docker image from source and add the Intel Media SDK, VA-API and/or whatever else is required and give that a go.

Of course, the first step is pulling the ARM project and building the ARM image and the ARM-dependencies image to make sure that's working, but I cant get those two builds to work, (ARM and/or arm-dependencies).

Without digging too deeply, the problem seems to be centered on the arm-dependencies.  On the github page for arm-dependencies, there is a RED failing error on the Deploy Docker Image icon and clicking on that shows a series of fail error messages for the last week or so.

Before I spin my wheels any further, is there a known problem with building arm-dependencies currently?  Is there a previous version I can fall back to that will build successfully?

I'm just curious, and apologize if I've missed any status postings somewhere related to this issue.

I've built the Intel Media SDK and the VA-API support with my previous efforts and thought I could get something together, and who knows, it might help someone else.

Hope to hear back and thanks!

**Comment by DZaremba14600** [comment]
To add to the previous post, are there any suggestions on building this image that I may have  missed?
Should I build 'arm-dependencies' first as its own stand alone image?  Can I build that as is? or is it preferred to pull automatic-ripping-machine, merge in arm-dependencies and build that all at once?  Did I mess something up with the requirements.txt modification mentioned on the arm-dependencies page in the instructions?  (just add that one line and one line alone to the root requirements.txt in automatic-ripping-machine?

Any clarifications would be great.

Thanks again.

**Reply by microtechno9000** [reply]
Thanks will check arm dependencies 

As for the other docker build bits, will check and get back to you

**Comment by shitwolfymakes** [comment]
Troubleshooting now, standby

**Reply by shitwolfymakes** [reply]
Fix has been merged with https://github.com/automatic-ripping-machine/arm-dependencies/pull/301

**Comment by DZaremba14600** [comment]
Thank you.

I took a look at the change you made via the link you reference, and that is the module I was getting the error at.  I'll pull this project again this evening and see if I can build the Docker image without any errors.  I've got some Docker experience, but its been a while since I created my own DockerFile from scratch and built an image from it.  Since your fix references the module I was getting an error on, I'd assume this will build fine.  From there I'll try the few things I wanted to experiment with.  If it works, I'll be sharing them in the hope they're of benefit to someone.

Thanks again for your help on this!

**Reply by DZaremba14600** [reply]
Thanks for the fix.  I pulled the project and built it successfully.

I also did what I want to do regarding QSV and had success! 

QSV working in the docker container.  I've got a quick temp fix and a fix for the Docker Build.

I'll be posting a new thread about that.

Hope it helps someone.

Thanks.

**Reply by DZaremba14600** [reply]
As I stated, I'm no Docker expert.

I pulled the project and built it.  It went fine.  I absolutely wanted to make sure I wasn't pulling arm-dependencies from Docker Hub, that I was truly building the project.  I ended up building arm-dependencies standalone and using a tag of :daz and changing the root DockerFile FROM xxx AS base to refer to the tagged arm-dependencies:daz and ensure I was building the submodule image and the main image totally from scratch.

That all worked fine, no errors. 

Once that was the case I began my experimenting and testing.

I'll follow up with that in a different thread.

But thanks for identifying the issue.
